text/0077-issue-platform.md

```diff
@@ -0,0 +1,85 @@
+  - An id of an existing event. This is mostly to handle events that are already received by sentry and stored in nodestore, like errors and transactions
+  - A full event payload. In this case, the detector fits their data to the event format as best as they can and sends it through. We set the type of this event to “generic”, then save it via `EventManager`, so it ends up in nodestore.
+
+Once we’ve ingested the occurrence and optionally the event, we then create/update the issue based on the fingerprint (note: currently, we just grab the first fingerprint out of the list and use that, but this can be expanded).
+
+We then write the event to Snuba via Eventstream. This is written to our own dataset designed to handle a wide range of different events.
+
+## Search
+Search remains powered by Snuba. The main difference is that we’re making 3 separate queries, one to errors, one to transactions (performance issues) and one to the issue platform. We join these results and combine with the group table to produce the issue list.
+
+This will be unified into a query to errors and a query to the issue platform dataset, once performance issues have been migrated over.
+
+## Issue Details
+This page remains largely the same. The main difference is that we return the occurrence data with the event, which is used to display the `evidence_display` data by default. Frontend teams might also customize the display of the page for an issue type using `evidence_data`.
+
+## Creating a new issue type
+To create a new issue type the team just implements a subclass with the relevant data (https://github.com/getsentry/sentry/blob/5fd8242aca575c84314f1b6744be2ec1d2e9e471/src/sentry/issues/grouptype.py#L151-L156).
+
+Originally, we planned to make this dynamic. For now, we're going to leave this requiring a pr, but if we do decide want to make this dynamic in the future it won't be a huge undertaking.
+
+
+# Future Plans:
+- Port performance issues to the issue platform. This gives us more consistency, and will improve query performance since we no longer need to query the transactions table, which contains mostly irrelevant data.
+- Improve the release process for new issue types. This is via a policy layer, that will automatically set up ways to release new issue types to LA, EA and GA and save adding a bunch of boilerplate code.
+
+# Things we’ve cut:
+Errors. We originally had plans to send errors through the issue platform to centralise everything. We’ve decided against this because it seems like a high risk task without much reward. The primary reason we wanted to send errors through the issue platform was so that we could split up detection from ingestion. We might still want to separate detection from ingestion in the future, and this doesn't necessarily have to involve the issue platform, or could involve a separate path through issue platform ingestion that still writes to the same errors dataset.
```

Comment:
> The primary reason we wanted to send errors through the issue platform was so that we could split up detection from ingestion.

Does it matter much for errors? Each event is also an occurrence so there is no real difference between the two.

Reply:
The plan now however from my understanding is to move errors onto issue platform. This is something I am at least very interested in as it would allow us to test different versions of grouping algorithms without impacting existing groups.

Reply:
Can we keep the storage where it is? that does not impact the grouping algorithm.

Reply:
@fpacifici I'm not involved with the project to move errors to issue platform. There are two things we definitely need to do:

a) we need to enable that an error can exist in two issues. That the existing errors data set does not really support due to it's n:1 relationship
b) from a programming point of view we need to have a reasonable level of abstraction between all the issue types to drive future features such as detecting issue state.

The general assumption that I had is that everything moves over to this new issue world.